lib/pry/input_completer.rb

```diff
@@ -239,4 +222,47 @@ def build_path(input)
     end
     return path, input
   end
+
+  private
+
+  def all_available_methods_cached
+    current_version = methods_cache_version
+    unless current_version && @methods_cache_version == current_version
+      @all_available_methods_cached = all_available_methods
+      @methods_cache_version = current_version
+    end
+    @all_available_methods_cached
+  end
+
+  # For MRI we simply use RubyVM.stat, for other rubies we count
+  # all defined instance methods. While we still iterate over ObjectSpace
+  # we can have improvements from not sorting methods list.
```

Comment:
Are you sure that sorting is the slow part?

In my case (using MRI), and with my branch, the ObjectSpace scan takes like 110ms, and the sort+uniq took just 45ms. Moving `uniq!` forward to be called first turned that into 30ms.